### DISCUSSION TITLE
Permission error preventing the use of DeDRM

### DISCUSSION THREAD

**Discussion by jomteon** [noDRM/DeDRM_tools#502] [Q&A]
I'm using DeDRM 10.0.9 (although I did also try 7.2.1) and trying to get it working on Calibre.  

When I click the configuration option "Kindle for Android ebooks" and choose a backup file, I get a message that says it cannot open it as a temp file is already running. The full error is:.

```
calibre, version 7.5.1
ERROR: Cannot open file or folder as it is in use: <p>Could not open the file: "C:\Users\MyName\AppData\Local\Temp\tmprh6ng5qt". It is already opened in the following programs:<div>The main calibre program: C:\Program Files\Calibre2\calibre.exe

calibre 7.5.1  embedded-python: True
Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19045')
Python 3.11.5
Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Interface language: None
EXE path: C:\Program Files\Calibre2\calibre.exe
Successfully initialized third party plugins: DeDRM (10, 0, 9)
Traceback (most recent call last):
  File "C:\Users\MyName\AppData\Roaming\calibre\plugins\DeDRM.zip\config.py", line 1424, in get_android_file
  File "C:\Users\MyName\AppData\Roaming\calibre\plugins\DeDRM.zip\androidkindlekey.py", line 251, in get_serials
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 
C:\\Users\\MyName\\AppData\\Local\\Temp\\tmprh6ng5qt'
````

Things I've done to remedy this (based largely on results from somewhat related help threads):

- rebooting, of course
- going into the Windows 10 environment variables. I tried both "user variables" and "system variables" clicking on New and setting the Variable to CALIBRE_TEMP_DIR and the Value to a different folder to hold temp files. (I rebooted after this as well)
- I also tried to change the temp folder by starting up calibre with a batch file, using the line

`set CALIBRE_TEMP_DIR=D:\Software\calibre_temp\Temporary\Calibre\temp`

None of this has worked. Not only has it not fixed the issue, it seems that my adjustments to the environment variables didn't do anything either. Calibre is still creating the troublesome files in C:\\Users\\MyName\\AppData\\Local\\Temp, and I'm not even sure if that's the actual issue or not in the end.

Not sure how to proceed from here.

edit: Also tried running all this in Safe Mode. Got this addition to the error:
`ERROR: Cannot open file or folder as it is in use: The file "C:\Users\MyName\AppData\Local\Temp\tmpe8gjs9ji" is opened in another program, so calibre cannot access it.This is usually caused by software such as antivirus or file sync (aka DropBox and similar) accessing files in the calibre library folder at the same time as calibre. Try excluding the calibre library folder from such software.`

However, as can be seen in the initial error message, the problem is that:
`It is already opened in the following programs:<div>The main calibre program: C:\Program Files\Calibre2\calibre.exe`
Calibre itself is apparently the problem.

Edit 2: Following advice in [this thread](https://www.mobileread.com/forums/showthread.php?p=4400996), I renamed the temp file that was generated as map_data_storage.db and applied that to "Kindle for Android ebooks". Then I grabbed the prc file and copied it to calibre and tried to convert it. However it failed, telling me that it was DRM protected.

**Comment by ElleKayEm** [comment]
Try this auto-release version of DeDRM: https://github.com/noDRM/DeDRM_tools_autorelease/releases/tag/autorelease_bf2471e65b1f52bb5292caeba70a9aea31bf6653

It has the latest changes and includes a fix for this.
